analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/ExecutingSqlQueriesBase.cs

```diff
@@ -95,7 +95,22 @@ public abstract class ExecutingSqlQueriesBase<TSyntaxKind, TExpressionSyntax, TI
 
         private readonly MemberDescriptor[] invocationsForFirstArgument =
             {
-                new(KnownType.System_Data_Sqlite_SqliteCommand, "Execute")
+                new(KnownType.System_Data_Sqlite_SqliteCommand, "Execute"),
+                new(KnownType.NHibernate_ISession, "CreateQuery"),
+                new(KnownType.NHibernate_ISession, "CreateSQLQuery"),
+                new(KnownType.NHibernate_ISession, "Delete"),
+                new(KnownType.NHibernate_ISession, "DeleteAsync"),
+                new(KnownType.NHibernate_ISession, "GetNamedQuery"),
+                new(KnownType.NHibernate_Impl_SessionImpl, "CreateQuery"),
+                new(KnownType.NHibernate_Impl_SessionImpl, "CreateSQLQuery"),
+                new(KnownType.NHibernate_Impl_SessionImpl, "Delete"),
+                new(KnownType.NHibernate_Impl_SessionImpl, "DeleteAsync"),
+                new(KnownType.NHibernate_Impl_SessionImpl, "GetNamedQuery"),
+                new(KnownType.NHibernate_Impl_SessionImpl, "GetNamedSQLQuery"),
+                new(KnownType.NHibernate_Impl_AbstractSessionImpl, "CreateQuery"),
+                new(KnownType.NHibernate_Impl_AbstractSessionImpl, "CreateSQLQuery"),
+                new(KnownType.NHibernate_Impl_AbstractSessionImpl, "GetNamedQuery"),
```

Comment:
I'm not sure why this is needed, because `CreateFilter` is picked up from the interface, but removing this breaks the UTs

Reply:
It's because the inheritance hierarchy of NHibernate is a bit *odd* and our implementation doesn't do the right thing when it comes to interface implementations. To do it the right way we need to do something like described here:
https://stackoverflow.com/questions/42074792/roslyn-is-symbol-implementation-of-interface
This would be the correct thing to do, but it is more expensive also. As long as we can define the members explicitly (as here) and we don't get FN reported by users (e.g. if they implement interfaces we defined in a complex way, we don't recognize the methods anymore) we should leave it as is IMO.